### PR TITLE
Pathauto settings on user, term, node type forms.

### PR DESCRIPTION
**Pull request by jenlampton**
PR for https://github.com/backdrop/backdrop-issues/issues/899

Replaces https://github.com/backdrop/backdrop/pull/917

### CODE REVIEW

File: core/modules/path/path.inc [hunk 1]
```diff
@@ -382,7 +382,13 @@ function path_generate_entity_alias(Entity $entity, $source = NULL, $langcode =
...
   $entity_type = entity_get_info($entity_type_name);
   $token_type = isset($entity_type['token type']) ? $entity_type['token type'] : NULL;
-  $bundle = $entity->bundle();
+
+  if ($entity_type_name == 'taxonomy_term') {
+    $bundle = $entity->vocabulary;
+  }
+  else {
+    $bundle = $entity->bundle();
+  }
```
Comment (quicksketch) [inline core/modules/path/path.inc]: This indicates a bug in the TaxonomyTerm class. $entity->bundle() should work on all entities. We need to add this to TaxonomyTerm:

```
  /**
   * Implements EntityInterface::bundle().
   */
  public function bundle() {
    return $this->vocabulary;
  }
```

File: core/modules/taxonomy/taxonomy.admin.inc [hunk 2]
```diff
@@ -129,6 +133,26 @@ function taxonomy_form_vocabulary($form, &$form_state, TaxonomyVocabulary $vocab
     '#value' => '0',
   );
 
+  $form['path'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('URL pattern'),
+    '#collapsible' => TRUE,
+    '#collapsed' => TRUE,
+    '#group' => 'additional_settings',
+  );
```
Comment (quicksketch) [inline core/modules/taxonomy/taxonomy.admin.inc]: This needs to be wrapped in `if (module_exists('path')) {`

File: core/modules/node/node.types.inc [hunk 3]
```diff
@@ -263,6 +267,25 @@ function node_type_form($form, &$form_state, $type = NULL) {
     '#default_value' => $type->settings['node_submitted'],
     '#description' => t('Author username and publish date will be displayed.'),
   );
+  $form['path'] = array(
```
Comment (quicksketch) [inline core/modules/node/node.types.inc]: This needs to be wrapped in `if (module_exists('path')) {`

File: core/modules/taxonomy/taxonomy.admin.inc [hunk 4]
```diff
@@ -95,8 +95,12 @@ function taxonomy_form_vocabulary($form, &$form_state, TaxonomyVocabulary $vocab
     }
     $form_state['vocabulary'] = $vocabulary;
   }
-  else {
-    $vocabulary = $form_state['vocabulary'];
+
+  // Get the default term url pattern, if set.
+  $url_pattern = config_get('path.settings', 'taxonomy_term_pattern');
```
Comment (quicksketch) [inline core/modules/taxonomy/taxonomy.admin.inc]: This needs to be wrapped in `if (module_exists('path')) {`

File: core/modules/user/user.admin.inc [hunk 5]
```diff
@@ -302,9 +302,29 @@ function user_admin_settings($form, &$form_state) {
     '#description' => t("This text is displayed at the picture upload form in addition to the default guidelines. It's useful for helping or instructing your users."),
   );
 
+  $form['path'] = array(
```
Comment (quicksketch) [inline core/modules/user/user.admin.inc]: This needs to be wrapped in `if (module_exists('path')) {`

**Comment by jenlampton** [comment]
Pushed a fix for the failing test.

**Comment by quicksketch** [comment]
I manually merged this in with the minor feedback at bb82360. Thanks!

File: core/modules/path/path.module [hunk 6]
```diff
@@ -178,6 +219,104 @@ function path_form_taxonomy_form_term_alter(&$form, $form_state) {
...
+  if (isset($form_state['values']['path_settings'])) {
+    config_set('path.settings', 'user_pattern', $form_state['values']['path_settings']);
+  }
+}
+
+/**
+ * Return URL pattern settings form for easy addition into other forms.
+ *
+ * @param $settings
+ *   Array of variabels needed to complete the form, including:
```
Comment (klonos) [inline core/modules/path/path.module:287]: typo: "variabels"

Comment (jenlampton) [inline core/modules/path/path.module:287]: how do you see things like this?  Magic eyes :) Can you open a new issue? Maybe tag it novice?

**Comment by klonos** [comment]
> how do you see things like this? Magic eyes :)

:smile: ...as I often repeat in my posts, I have self-diagnosed OCD :stuck_out_tongue_winking_eye:. I admit that I even secretly correct the small typos you guys happen to make on your posts since I have permission to edit :smiling_imp: :shame: :innocent: 

Now, I do not code (no php or js - just html and css) but we did have some programming classes while I was doing my Diploma in IT. I started with Pascal (in DOS!!! ...yes, I'm old) and also did a little bit of C/C++ and VB and Java. No major/complex projects or anything - just the basics, but I like to believe that I have a grasp of basic programming (variables, operators, loops etc.) and given the time, I could pick up any programming language. Deep down I am a coder (wannabe). Anyways, point is that although I usually do top-level reviews (UI/UX), I always take a minute to quickly look at the code each time you guys do a PR or merge something.

My favorite image to post along with my comments each time I find something trivial like this with each amazing feature that gets implemented:

missed_a_spot

> Can you open a new issue? Maybe tag it novice?

I think someone [fixed it already](https://github.com/backdrop/backdrop/blob/1.x/core/modules/path/path.module#L302) (@quicksketch ??).

**Pull request closed without merging**